### PR TITLE
LPD 38548 Add additional parameter restrictedFieldNames to ignore data when import Batch

### PR DESCRIPTION
**Pull request by movaldivia**
https://liferay.atlassian.net/browse/LPD-38548

**What**

Added an additional parameter _restrictedFieldNames_ that let clients to ignore data when they are doing an import. For example you can ignore the permissions of your imported data.

### CODE REVIEW

**Comment by movaldivia** [comment]
ci:test:sf

**Comment by movaldivia** [comment]
ci:test:sf

**Comment by movaldivia** [comment]
ci:test:relevant

**Comment by movaldivia** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
Hey @movaldivia ,

it seems the failures could be related. I see some batch engine failures. Can you review?

headless-batch-engine-test:compileTestIntegrationJava

File: modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/vulcan/openapi/contributor/APIApplicationOpenApiContributor.java [hunk 1]
```diff
@@ -73,6 +74,10 @@ public void contribute(OpenAPI openAPI, OpenAPIContext openAPIContext)
 			return;
 		}
 
+		if (!FeatureFlagManagerUtil.isEnabled("LPD-35944")) {
```
Comment (ccorreagg) [inline modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/vulcan/openapi/contributor/APIApplicationOpenApiContributor.java:77]: This code is in the Headless Builder module. This is not related to the Headless Builder (also called API Builder), so this code should not be here.

If there are no OpenAPIContributors in the Batch modules, you must create one

**Comment by movaldivia** [comment]
ci:test:sf

**Comment by movaldivia** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
@movaldivia The SF is failing, and the relevant test, for any reason, it was not executed properly. Please, check it locally and fix it before pushing and/or executing any other test suite. Some errors make me suspect about to the code (SF and previously an error related to batch)

**Comment by ccorreagg** [comment]
We need to keep pushing this PR. In order to execute the SF in the `portal.properties` file, please execute `ant format-source-current-branch` from the `portal-impl` folder.

**Comment by movaldivia** [comment]
@ccorreagg yes the thing is that my changes are not here.

Figuring out now how to fix it

image

**Comment by movaldivia** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
image

**Comment by movaldivia** [comment]
@ccorreagg 

image
 I get this D:

**Comment by movaldivia** [comment]
ci:test:sf

**Comment by movaldivia** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
Hey @movaldivia ,

I have included several commits, please, review them. I saw the Feature Flag is not the epic ticket, so we need to update the code to make it match. After that, I will review the tests

**Comment by ccorreagg** [comment]
By the way, I see this confusing commit:

```
commit 8a0c06023c26d852bfb06fc48664a1053f8d2d5e
Author: movaldivia <movaldivia@uc.cl>
Date:   Tue Nov 12 16:00:02 2024 +0100

    LPD-38548 fix tests

modules/apps/batch-engine/batch-engine-service/src/main/java/com/liferay/batch/engine/internal/reader/BatchEngineImportTaskItemReaderUtil.java
modules/apps/headless/headless-batch-engine/headless-batch-engine-impl/src/main/java/com/liferay/headless/batch/engine/internal/vulcan/openapi/contributor/BatchEngineOpenApiContributor.java
modules/apps/headless/headless-batch-engine/headless-batch-engine-test/src/testIntegration/java/com/liferay/headless/batch/engine/resource/v1_0/test/ExportTaskResourceTest.java
modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/vulcan/openapi/contributor/APIApplicationOpenApiContributor.java

```

You are just fixing one test but the rest of the files are not related to the test. This would be really confusing if we have to backport the story. Please, could you split it into different commits?

**Comment by movaldivia** [comment]
@ccorreagg 

Did not understart  this [commit](https://github.com/liferay-headless/liferay-portal/pull/2330/commits/ed83847a6895db2c00d82f763fd3758d9e22c0b6) why with the previous implementation the method was being callend more than once?

In this [commit](https://github.com/liferay-headless/liferay-portal/pull/2330/commits/bca1636d0082152c175ed00d30bac826c421ab5d)  is better to return a Collections.emptySet();?

Great utils [commit](https://github.com/liferay-headless/liferay-portal/pull/2330/commits/18501522cb37ddc4910135a1e2654fc870c6ec03)

I think that the last [commit](https://github.com/liferay-headless/liferay-portal/pull/2330/commits/0e5a00ec1475b881da10d8c3a159bdf60de60156) breaks the expected behaviour 

image
image
image

Confirm me to start the fixes

**Comment by ccorreagg** [comment]
> @ccorreagg
> 
> Did not understart this [commit](https://github.com/liferay-headless/liferay-portal/pull/2330/commits/ed83847a6895db2c00d82f763fd3758d9e22c0b6) why with the previous implementation the method was being callend more than once?
> 
> In this [commit](https://github.com/liferay-headless/liferay-portal/pull/2330/commits/bca1636d0082152c175ed00d30bac826c421ab5d) is better to return a Collections.emptySet();?
> 
> Great utils [commit](https://github.com/liferay-headless/liferay-portal/pull/2330/commits/18501522cb37ddc4910135a1e2654fc870c6ec03)
> 
> I think that the last [commit](https://github.com/liferay-headless/liferay-portal/pull/2330/commits/0e5a00ec1475b881da10d8c3a159bdf60de60156) breaks the expected behaviour
> 
> image image image
> 
> Confirm me to start the fixes

About the first question, the `FeatureFlagManagerUtil.isEnabled("LPD-35944")` method is being called for each iteration over the `for (Map.Entry<String, Object> entry : fieldNameValueMap.entrySet()) {` loop. For that reason, I moved that to the private method to do it just once.

About the second question, you were setting an empty map to the `parameters` attribute, and then, keep executing some code over the `parameters` map. I think it is faster just to return the empty set directly. I think it is the same calling to the method or creating the HashSet directly.

About the third question, you are right. I think I broke it. I updated with a newer commit, please check and if it keeps broken, let me know and I will test it until I leave it fixed

File: modules/apps/headless/headless-batch-engine/headless-batch-engine-impl/src/main/java/com/liferay/headless/batch/engine/internal/vulcan/openapi/contributor/BatchEngineOpenApiContributor.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+import java.util.List;
+import java.util.Objects;
+
+@Component(service = OpenAPIContributor.class)
+public class BatchEngineOpenApiContributor implements OpenAPIContributor {
+	@Override
+	public void contribute(OpenAPI openAPI, OpenAPIContext openAPIContext)
+		throws Exception {
+		System.out.println("WENA PERRO");
```
Comment (ccorreagg) [inline modules/apps/headless/headless-batch-engine/headless-batch-engine-impl/src/main/java/com/liferay/headless/batch/engine/internal/vulcan/openapi/contributor/BatchEngineOpenApiContributor.java]: Hey, please, get rid of this. We should not send this to the repo

**Comment by movaldivia** [comment]
ci:test:sf

**Comment by movaldivia** [comment]
ci:test:relevant

**Comment by movaldivia** [comment]
ci:test:sf

**Comment by movaldivia** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by movaldivia** [comment]
ci:test:sf

**Comment by movaldivia** [comment]
ci:test:relevant

**Comment by movaldivia** [comment]
@ccorreagg there are two types of feature flags system and instance based. Our is instance based and works as expected. To make it system based we would have needed to follow this example: `feature.flag.LPS-187285.system=true`

**Comment by ccorreagg** [comment]
ci:forward

**Comment by ccorreagg** [comment]
hey @movaldivia please, remember to monitor the PRs when the tests finish to check if there is something related, to forward or to re-execute the tests

**Comment by movaldivia** [comment]
ci:help
